parquet-tools/src/main/java/org/apache/parquet/tools/command/CatCommand.java

```diff
@@ -71,14 +75,19 @@ public void execute(CommandLine options) throws Exception {
     try {
       PrintWriter writer = new PrintWriter(Main.out, true);
       reader = ParquetReader.builder(new SimpleReadSupport(), new Path(input)).build();
+      ParquetMetadata metadata = ParquetFileReader.readFooter(new Configuration(), new Path(input));
+      JsonRecordFormatter.JsonGroupFormatter formatter = JsonRecordFormatter.fromSchema(metadata.getFileMetaData().getSchema());
+
       for (SimpleRecord value = reader.read(); value != null; value = reader.read()) {
         if (options.hasOption('j')) {
-          value.prettyPrintJson(writer);
+          writer.write(formatter.formatRecord(value));
```

Comment:
The primary issue is that `SimpleObject` doesn't understand the concept of an array. It simulates an array by adding multiple items to the same key. The problem with that approach is that the Json representation then depends on the amount of data written. A key with only one value would appear as a singleton-object, while items with multiple entries appended to the same key would appear as an array. This would result in an inconsistent json encoding.